core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -461,6 +462,7 @@ public synchronized boolean isReady() {
 
       @Override
       public void cancel(Status reason) {
+        reason = stripCause(reason);
```

Comment:
This is `ClientStream.cancel(..)` and calls `serverStreamListener.closed(Status)` ... surely that should be stripped? The `serverStreamListener.closed(Status)` would be called on the server after a RST_STREAM has been received, no?